packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,672 @@
+                transaction,
+                txOrigin,
+                transactionSignature,
+                approvalExpirationTimeSeconds,
+                approvalSignatures,
+                {
+                    from: txOrigin,
+                    gas: orderTransactionOpts.gasLimit,
+                    gasPrice: orderTransactionOpts.gasPrice,
+                    nonce: orderTransactionOpts.nonce,
+                },
+            );
+        }
+        const txHash = await this._contractInstance.executeTransaction.sendTransactionAsync(
+            transaction,
+            txOrigin,
+            transactionSignature,
+            approvalExpirationTimeSeconds,
+            approvalSignatures,
+            {
+                from: txOrigin,
+                gas: orderTransactionOpts.gasLimit,
+                gasPrice: orderTransactionOpts.gasPrice,
+                nonce: orderTransactionOpts.nonce,
+            },
+        );
+        return txHash;
+    }
+
+    private async _toSignedZeroExTransactionAsync(
+        data: string,
+        senderAddress: string,
+    ): Promise<SignedZeroExTransaction> {
+        await assert.isSenderAddressAsync('senderAddress', senderAddress, this._web3Wrapper);
+        const normalizedSenderAddress = senderAddress.toLowerCase();
+
+        const transaction: ZeroExTransaction = {
+            salt: generatePseudoRandomSalt(),
+            signerAddress: normalizedSenderAddress,
+            data,
+            verifyingContractAddress: this.exchangeAddress,
+        };
+        const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+        const signature = await signatureUtils.ecSignHashAsync(
+            this._web3Wrapper.getProvider(),
+            transactionHash,
+            transaction.signerAddress,
+        );
+        return {
+            ...transaction,
+            signature,
+        };
+    }
+
+    private async _requestServerResponseAsync(
+        transaction: SignedZeroExTransaction,
+        senderAddress: string,
+        feeRecipientAddress: string,
+    ): Promise<CoordinatorServerApprovalResponse | CoordinatorServerCancellationResponse> {
+        assert.doesConformToSchema('transaction', transaction, schemas.zeroExTransactionSchema);
+        assert.isETHAddressHex('feeRecipientAddress', feeRecipientAddress);
+
+        const coordinatorOperatorEndpoint = await this._registryInstance.getCoordinatorEndpoint.callAsync(
+            feeRecipientAddress,
+        );
+        if (coordinatorOperatorEndpoint === undefined) {
+            throw new Error(
+                `Could not find endpoint for coordinator operator with { coordinatorAddress: ${
+                    this.address
+                }, registryAddress: ${this.registryAddress}, operatorAddress: ${feeRecipientAddress}}`,
+            );
+        }
+
+        const requestPayload = {
+            signedTransaction: transaction,
+            txOrigin: senderAddress,
```

Comment:
`txOrigin` should default  to `takerAddress`. `txOrigin` specifies who can ultimately submit the Ethereum transaction to the network.

Reply:
This is passed in from any calling methods but the variable is named `txOrigin` for consistency with the request payload schema that the server expects. When called for a fill approval it's passed as the takerAddress, and when called for a soft cancellation it's the makerAddress. Is that right?